### PR TITLE
ACMS-1499: Refactored permission issue and fixed PHPunit test cases

### PR DESCRIPTION
**Pull request by deepakmishra2**
**Motivation**

Fixes ACMS-1499

**Proposed changes**
Refactoring permission so that on install permission is assigned to those roles

**Alternatives considered**

**Testing steps**

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_common/config/install/user.role.user_administrator.yml [hunk 1]
```diff
@@ -6,9 +6,6 @@ label: 'User Administrator'
 weight: -4
 is_admin: null
 permissions:
-  - 'administer CAPTCHA settings'
```
Comment (rajeshreeputra) [inline modules/acquia_cms_common/config/install/user.role.user_administrator.yml:9]: these permission got removed from here but not added conditional basis, please update accordingly.

File: modules/acquia_cms_common/tests/src/Functional/MediaPermissionsTestBase.php [hunk 2]
```diff
@@ -42,9 +42,6 @@ public function providerBasicPermissions(): array {
       [
         'user_administrator',
         [
-          'administer CAPTCHA settings',
-          'administer honeypot',
```
Comment (rajeshreeputra) [inline modules/acquia_cms_common/tests/src/Functional/MediaPermissionsTestBase.php:46]: these permission needs to checked here based on module availability

File: modules/acquia_cms_site_studio/acquia_cms_site_studio.module [hunk 3]
```diff
@@ -174,15 +174,17 @@ function acquia_cms_site_studio_rebuild_styles() {
...
-  if (in_array($role->id(), [
-    'content_administrator',
-    'content_editor',
-    'content_author',
-    'site_builder',
-  ])) {
-    $permissions = SiteStudioPermissionHelper::getSiteStudioPermissionsByRole($role->id());
-    foreach ($permissions as $permission) {
-      $role->grantPermission($permission);
+  if (\Drupal::configFactory()->getEditable('pack_acquia_cms_article')) {
```
Comment (rajeshreeputra) [inline modules/acquia_cms_site_studio/acquia_cms_site_studio.module:177]: Why are we checking this non existing config here

File: modules/acquia_cms_common/tests/src/fixtures/permissions/basic/user.role.user_administrator.yml [hunk 4]
```diff
@@ -1,7 +1,4 @@
 permissions:
-  - 'administer CAPTCHA settings'
-  - 'administer honeypot'
```
Comment (rajeshreeputra) [inline modules/acquia_cms_common/tests/src/fixtures/permissions/basic/user.role.user_administrator.yml:3]: these permission needs to checked here based on module availability

File: tests/src/fixtures/permissions/integrated/user.role.user_administrator.yml [hunk 5]
```diff
@@ -12,9 +12,6 @@ weight: -4
 is_admin: null
 permissions:
   - 'access toolbar'
-  - 'administer CAPTCHA settings'
```
Comment (rajeshreeputra) [inline tests/src/fixtures/permissions/integrated/user.role.user_administrator.yml:15]: these permission needs to checked here based on module availability

**Review by rajeshreeputra — APPROVED** [review]
Changes looks good to me.

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**